Common/Securities/Option/Option.cs

```diff
@@ -435,8 +436,32 @@ public void SetFilter(Func<OptionFilterUniverse, OptionFilterUniverse> universeF
         /// <param name="universeFunc">new universe selection function</param>
         public void SetFilter(PyObject universeFunc)
         {
-            var pyUniverseFunc = PythonUtil.ToFunc<OptionFilterUniverse, OptionFilterUniverse>(universeFunc);
-            SetFilter(pyUniverseFunc);
+            ContractFilter = new FuncSecurityDerivativeFilter(universe =>
+            {              
+                var optionUniverse = universe as OptionFilterUniverse;
+                using (Py.GIL())
+                {
+                    PyObject result = (universeFunc as dynamic)(optionUniverse);
+
+                    //Try to convert it to the possible outcomes and process it
+                    List<Symbol> list;
+                    OptionFilterUniverse filter;
+
+                    if ((result).TryConvert(out list))
+                    {
+                        optionUniverse = optionUniverse.WhereContains(list);
+                    }
+                    else if ((result).TryConvert(out filter))
+                    {
+                        optionUniverse = filter;
+                    }
+                    else
+                    {
+                        throw new ArgumentException($"QCAlgorithm.AddUniverse: {universeFunc.Repr()} is not a valid argument.");
```

Comment:
`**QCAlgorithm.AddUniverse**: {universeFunc.Repr()} is not a valid argument.` not sure this message is right? Maybe we could also add expected valid arguments for `SetFilter`